### PR TITLE
feat: add new preset and preset utils

### PR DESCRIPTION
**Pull request by targeral**
## Description

add new buildPreset and utils function

## Types of changes

- [x] Docs change / Dependency upgrade
- [ ] Bug fix
- [x] New feature / Improvement
- [ ] Refactoring
- [ ] Breaking change

## Checklist

- [x] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/toolkit/module-doc/docs/zh/api/config/build-preset.mdx [hunk 1]
```diff
@@ -232,3 +232,75 @@ export default defineConfig({
...
+  buildPreset({ preset }) {
+    return preset['npm-library'];
+  },
+});
+```
+---
+
+使用 `buildPreset` 所支持的字符串。
+
+```ts modern.config.ts
```
Comment (jkzing) [inline packages/toolkit/module-doc/docs/zh/api/config/build-preset.mdx:265]: 这里是忘写了吗..

Comment (targeral) [reply packages/toolkit/module-doc/docs/zh/api/config/build-preset.mdx:265]: 不是，这里是空的内容的话，就是默认使用的是上面部分代码显示的内容。这个好像是 CH.Spotlight 一个特性。不过我这里补充一下，这样看起来不会太迷惑。

Comment (targeral) [reply packages/toolkit/module-doc/docs/zh/api/config/build-preset.mdx:265]: 已更新

Comment (jkzing) [reply packages/toolkit/module-doc/docs/zh/api/config/build-preset.mdx:265]: 是我孤陋寡闻了...😂

**Review by jkzing — APPROVED** [review]

**Pull request merged**